mbed_lstools/lstools_linux_generic.py

```diff
@@ -88,10 +88,10 @@ def list_mbeds(self):
         self.ERRORLEVEL_FLAG = 0
 
         result = []
-        tidhex = re.compile(r'_([0-9a-fA-F]+)-\d+:\d+')
+        tidpattern = re.compile(r'_([0-9a-zA-Z]+)-\d+:\d+')
```

Comment:
Note: four first characters define [manufacturer_ids](https://github.com/ARMmbed/mbed-ls/blob/master/mbed_lstools/lstools_base.py#L44).
If you wish to detect correctly your devices, e.g. [XPRO_SAM](https://github.com/ARMmbed/mbed-ls/blob/master/mbed_lstools/lstools_base.py#L96) you have to make sure that your target id in USB ID starts with a proper substring, you XPRO case e.g. ```0900``` will be mapped to ```XPRO_SAMR21```.
So any target id like '''0900deadcodedeadcodedeadcode``` will work!